src/main/java/org/truffleruby/interop/RubyMessageResolution.java

```diff
@@ -278,4 +278,15 @@ protected Object access(VirtualFrame frame, DynamicObject receiver, String name,
 
     }
 
+    @Resolve(message = "NEW")
+    public static abstract class ForeignNewNode extends Node {
+
+        @Child private DispatchHeadNode dispatchHeadNode = insert(new DispatchHeadNode(true, false, MissingBehavior.CALL_METHOD_MISSING, DispatchAction.CALL_METHOD));
```

Comment:
The ForeignInvokeNode above should use `CallDispatchHeadNode.createOnSelf()` and also without `insert()`.